iac_controller.py

```diff
@@ -0,0 +1,36 @@
+import os
+from collections import deque
+from typing import Collection
+
+import requests
+from requests.auth import HTTPBasicAuth
+
+
+class IACController:
```

Comment:
Liking the design of this one 👍 

Fill pool (with a request to the IAC service) to drain from then refill when empty. Nice.